From 52e895fea3a2c10939ece49fa88cba89917b354b Mon Sep 17 00:00:00 2001
From: uros <uros@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 25 Jul 2014 07:47:36 +0000
Subject: [PATCH] 	Backport from mainline
 	2014-07-14  Jakub Jelinek  <jakub@redhat.com>

	PR target/61656
	* config/i386/i386.c (classify_argument): Don't merge classes above
	number of words.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_9-branch@213046 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/config/i386/i386.c |    2 +-
 2 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/gcc/config/i386/i386.c b/gcc/config/i386/i386.c
index 6dc6888..0a6e644 100644
--- a/gcc/config/i386/i386.c
+++ b/gcc/config/i386/i386.c
@@ -6552,7 +6552,7 @@ classify_argument (enum machine_mode mode, const_tree type,
 					   bit_offset);
 		  if (!num)
 		    return 0;
-		  for (i = 0; i < num; i++)
+		  for (i = 0; i < num && i < words; i++)
 		    classes[i] = merge_classes (subclasses[i], classes[i]);
 		}
 	    }
-- 
1.7.1

